vars/seleniumExecuteTests.groovy

```diff
@@ -9,28 +10,70 @@ import groovy.text.SimpleTemplateEngine
 
 @Field String STEP_NAME = getClass().getName()
 
-@Field GENERAL_CONFIG_KEYS = STEP_CONFIG_KEYS
-
-@Field Set STEP_CONFIG_KEYS = [
-    'buildTool', //defines the tool which is used for executing the tests
-    'containerPortMappings', //port mappings required for containers. This will only take effect inside a Kubernetes pod, format [[containerPort: 1111, hostPort: 1111]]
-    'dockerEnvVars', //envVars to be set in the execution container if required
-    'dockerImage', //Docker image for code execution
-    'dockerName', //name of the Docker container. This will only take effect inside a Kubernetes pod.
-    'dockerWorkspace', //user home directory for Docker execution. This will only take effect inside a Kubernetes pod.
+//TODO: limit parameter visibility
+@Field GENERAL_CONFIG_KEYS = [
+    /**
+     * Defines the tool which is used for executing the tests
+     * @possibleValues `'maven'`, `'npm'`
+     */
+    'buildTool',
+    /** @see dockerExecute */
+    'containerPortMappings',
+    /** @see dockerExecute */
+    'dockerEnvVars',
+    /** @see dockerExecute */
+    'dockerImage',
+    /** @see dockerExecute */
+    'dockerName',
+    /** @see dockerExecute */
+    'dockerWorkspace',
+    /**
+     * With `failOnError` the behavior in case tests fail can be defined.
+     * @possibleValues `true`, `false`
+     */
     'failOnError',
-    'gitBranch', //only if testRepository is used: branch of testRepository. Default is master
-    'gitSshKeyCredentialsId', //only if testRepository is used: ssh credentials id in case a protected testRepository is used
-    'sidecarEnvVars', //envVars to be set in Selenium container if required
-    'sidecarImage', //image for Selenium execution which runs as sidecar to dockerImage
-    'sidecarName', //name of the Selenium container. If not on Kubernetes pod, this will define the name of the link to the Selenium container and is thus required for accessing the server, example http://selenium:4444 (default)
-    'sidecarVolumeBind', //volume bind. This will not take effect in Kubernetes pod.
-    'stashContent', //list of stash names which are required to be unstashed before test run
-    'testRepository' //if tests are in a separate repository, git url can be defined. For protected repositories the git ssh url is required
+    /**
+     * Only if `testRepository` is provided: Branch of testRepository, defaults to master.
```

Comment:
[minor] do we need to explain the defaulting behaviour? Default should be retrieved by the docu generator and displayed alongside with the parameter anyway.

Reply:
This is a default handled by the `git` step. So it won't show up in the docs. But maybe we should set this to `master` by default.